src/rdkafka_admin.c

```diff
@@ -2735,3 +2837,400 @@ rd_kafka_DescribeConfigs_result_resources (
+        struct rd_kafka_partition_leader *leader;
+
+
+        rd_list_init(&leaders, offsets->cnt,
+                     (void *)rd_kafka_partition_leader_destroy);
+        err = rd_kafka_topic_partition_list_query_leaders(rk, offsets, &leaders,
+                                                          timeout_ms);
+
+        if (err) {
+                rd_list_destroy(&leaders);
+                return err;
+        }
+
+        rd_kafka_AdminFanoutState_t *state =
+                rd_kafka_AdminFanoutState_new(rd_list_cnt(&leaders),
+                                              rd_kafka_topic_partition_list_free,
+                                              rd_kafka_topic_partition_list_free);
+
+        rd_kafka_topic_partition_list_t *results =
+                rd_kafka_topic_partition_list_new(offsets->cnt);
+        rd_list_add(&state->responses, results);
+
+        /* For each leader send a request for its partitions */
+        RD_LIST_FOREACH(leader, &leaders, i) {
+                rd_kafka_op_t *rko;
+
+                rko = rd_kafka_admin_request_op_new(rk,
+                                                    RD_KAFKA_OP_DELETERECORDS,
+                                                    RD_KAFKA_EVENT_DELETERECORDS_RESULT,
+                                                    &cbs, options, rkqu);
+                rko->rko_u.admin_request.broker_id = leader->rkb->rkb_nodeid;
+                rd_kafka_AdminOptions_set_opaque(&rko->rko_u.admin_request.options, state);
+                rd_list_init(&rko->rko_u.admin_request.args, 1,
+                             rd_kafka_topic_partition_list_free);
+                rd_list_add(&rko->rko_u.admin_request.args,
+                            rd_kafka_topic_partition_list_copy(leader->partitions));
+                rd_list_add(&state->requests,
+                            rd_kafka_topic_partition_list_copy(leader->partitions));
+                rd_kafka_q_enq(rk->rk_ops, rko);
+        }
+
+        rd_list_destroy(&leaders);
+        return RD_KAFKA_RESP_ERR_NO_ERROR;
+}
+
+
+/**
+ * @brief Get an array of topic and partition results from a DeleteRecords result.
+ *        The returned objects will contain \c topic, \c partition, and \c offset.
+ *        \c offset will be set to the low-watermark (smallest available offset
+ *        of all live replicas).
+ *
+ * The returned object's life-time is the same as the \p result object.
+ */
+const rd_kafka_topic_partition_list_t *
+rd_kafka_DeleteRecords_result_offsets (
+        const rd_kafka_DeleteRecords_result_t *result) {
+        const rd_kafka_topic_partition_list_t *offsets;
+        const rd_kafka_op_t *rko = (const rd_kafka_op_t *) result;
+        size_t cntp;
+
+        rd_kafka_op_type_t reqtype =
+                rko->rko_u.admin_result.reqtype & ~RD_KAFKA_OP_FLAGMASK;
+        rd_assert(reqtype == RD_KAFKA_OP_DELETERECORDS);
+
+        cntp = rd_list_cnt(&rko->rko_u.admin_result.results);
+
+        rd_assert(cntp == 1);
```

Comment:
Why 1 and not > 0?

Reply:
We use an `rd_list` to match the existing `admin_result` interface, but are really just accumulating results into a single `topic_partition_list` at the head of the `rd_list`.